api4/team_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api4
+
+import (
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+)
+
+func TestCreateTeam(t *testing.T) {
+	th := Setup().InitBasic()
+	Client := th.Client
+
+	team := &model.Team{Name: GenerateTestUsername(), DisplayName: "Some Team", Type: model.TEAM_OPEN}
+	rteam, resp := Client.CreateTeam(team)
+	CheckNoError(t, resp)
+
+	if rteam.Name != team.Name {
+		t.Fatal("names did not match")
+	}
+
+	if rteam.DisplayName != team.DisplayName {
+		t.Fatal("display names did not match")
+	}
+
+	if rteam.Type != team.Type {
+		t.Fatal("types did not match")
+	}
+
+	_, resp = Client.CreateTeam(rteam)
+	CheckBadRequestStatus(t, resp)
+
+	rteam.Id = ""
+	_, resp = Client.CreateTeam(rteam)
+	CheckErrorMessage(t, resp, "A team with that name already exists")
```

Comment:
I actually have a PR based off this one that changes it over to using IDs, I was just waiting on this to go in